src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -3990,6 +4037,29 @@ def create_functionapp(cmd, resource_group_name, name, storage_account, plan=Non
 
     return functionapp
 
+def _validate_cpu_momory_functionapp(cpu=None, memory=None):
+    # validate either both cup and momory are provided or none is provided. throw error otherwise
+    if cpu is None and memory is None:
+        return
+    
+    if cpu is not None and memory is None:
+        raise ArgumentUsageError("--memory input is required with --cpu. Please provide both or none.")
+    
+    if cpu is None and memory is not None:
+        raise ArgumentUsageError("--cpu input is required with --memory. Please provide both or none.")
+    
+    # validate that memory is number and ends with Gi
+    if memory is not None and not memory.lower().endswith("gi"):
+        raise ValidationError("--memory input should end with Gi. Please provide a correct value. e.g. 4.0Gi.")
+    
+    if memory is not None:
+        try:
+            float(memory[:-2])
+        except ValueError:
+            raise ValidationError("--memory input is not valid. Please provide a correct value. e.g. 4.0Gi.")
+        
+    return
```

Comment:
Can you run `azdev style appservice` and fix those linting issues as well? I am pretty sure it will complain about this return here since it is unnecessary.

Reply:
Thank you! I have addressed them.